### DISCUSSION TITLE
Are defaults for REST datasource binding made mandatory?

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [Budibase/budibase#6775] [Help] [answered]
After the latest upgrade, I am seeing that I need to have a default value in the definition of the bindings of the REST datasource.
Till the last upgrade, I could clear out the default boxes (i.e. empty) and they would work.

Now, I am forced to enter dummy data. Is this correct?
Just want to make sure I am not missing anything

**Comment by melohagan** [comment]
Hey @SantoshSrinivas79 

I haven't noticed the issue you're describing. 

Would you be able to provide some reproduction steps and screenshots please?

**Comment by SantoshSrinivas79** [comment]
Hi @melohagan .... this issue persists ... I am calling this REST data source from a form. Why does the definition of the REST datasource need to have mandatory default values? When I remove the defaults and bind the values from the form, the REST POST call from Budibase is being sent with EMPTY values. Please advise .... I dont want to send the POST requests with default values.

### THIS WORKS

<img width="1011" alt="Screenshot 2022-09-22 at 1 29 38 PM" src="https://user-images.githubusercontent.com/1036163/191691096-bb567790-9c10-4cb3-9e9a-3c22e1d197ad.png">

<img width="983" alt="Screenshot 2022-09-22 at 1 29 55 PM" src="https://user-images.githubusercontent.com/1036163/191691160-53f1add6-39d6-45c7-9083-eca441ea5628.png">

<img width="854" alt="Screenshot 2022-09-22 at 1 33 11 PM" src="https://user-images.githubusercontent.com/1036163/191691884-3a3a2703-7875-466c-b056-83ec611f7340.png">

### THIS DOES NOT WORK

<img width="988" alt="Screenshot 2022-09-22 at 1 30 39 PM" src="https://user-images.githubusercontent.com/1036163/191691322-58116d64-f2f9-4869-9e8c-f03172ea3460.png">

<img width="979" alt="Screenshot 2022-09-22 at 1 30 52 PM" src="https://user-images.githubusercontent.com/1036163/191691383-6bd05c40-dc86-4e82-8a62-0daf3e7ac001.png">

<img width="854" alt="Screenshot 2022-09-22 at 1 33 11 PM" src="https://user-images.githubusercontent.com/1036163/191691884-3a3a2703-7875-466c-b056-83ec611f7340.png">

**Comment by SantoshSrinivas79** [comment]
Sorry for the nudge ... @melohagan @shogunpurple ... could you pls let me know if this is by design or am I missing something?

Basically, I don't want to keep stuff like secrets in the query definitions. On one of the earlier versions, the default values were not mandatory. I tried on the latest version and this issue seems to exist.

**Reply by melohagan** [reply]
Hey @SantoshSrinivas79 

To clarify, in the scenario that works, is it sending with the default values, or with the correct values from your form? 

In the not working scenario, it is not picking up the values from form?

**Reply by SantoshSrinivas79** [reply]
Hi @melohagan, please see answers below:

1. Works scenario: It sends the form binding values correctly. (basically .. the default values are *required* in the REST calls. They get overridden with the form values. (_this is all correct, except that I want to leave the default value to blank_)

2. Not working scenario: Budibase isn't triggering the call the end-point at all. My recipient endpoint (webhook) isn't getting any request at all. So, it is failing in Budibase itself. (server side I'm guessing)

**Comment by melohagan** [comment, marked as answer]
Thanks for the confirmation @SantoshSrinivas79 

I've raised a bug here: https://github.com/Budibase/budibase/issues/7986

**Reply by Abd-Atif** [reply]
As you marked it solved as a bug [https://github.com/Budibase/budibase/issues/7986] but I'm still facing this issue.
how to get rid of defaults values ?
lemme explain if I don't enter any default (dummy) value and execute the query in POST REST API, it will only show all empty values even you are entering some of values in the request and if give any default (dummy) values it executes perfectly, so in my scenario I want all those values should be empty if user didn't passed any value to that specific fields.
kindly help

**Reply by Abd-Atif** [reply]
also I'm using Budibase 2.2.23 version.

**Reply by melohagan** [reply]
Hey @Abd-Atif The current version is **2.4.8**
